### ISSUE TITLE
Option to create stacks without OnFailure: DELETE for debugging

### ISSUE THREAD

**Issue by duboisj-illuminate** [serverless/serverless#8538]
### Use case description

It appears that AWS CF stacks are created with a parameter telling AWS to delete the stack if it fails to create, as in this log from the serverless framework: 

Serverless: Creating Stack...
Serverless: [AWS cloudformation 200 0.651s 0 retries] createStack({
  StackName: 'reportingservice-poc-test2-dev-duboisj-add-tests',
  OnFailure: 'DELETE',
  Capabilities: [ 'CAPABILITY_IAM', 'CAPABILITY_NAMED_IAM', [length]: 2 ],

Seeming to come from this portion of the serveless framework code (from aws/deploy/lib/createStack.js):

    const params = {
      StackName: stackName,
      OnFailure: 'DELETE',
      Capabilities: ['CAPABILITY_IAM', 'CAPABILITY_NAMED_IAM'],
      Parameters: [],
      TemplateBody: JSON.stringify(this.serverless.service.provider.coreCloudFormationTemplate),
      Tags: Object.keys(stackTags).map(key => ({ Key: key, Value: stackTags[key] })),
    };

This seems to make it difficult to debug stack creation when it fails.  In particular, I assume serverless is supposed to be providing al of the error information which AWS output during stack creation in its logs.  I'm not confident that it is in my case, and I'd like to have the option to see the (uncreated, broken) stack still present to try and debug why my stack won't spin up.

Is there a way to override the 'OnFailure: 'DELETE' parameter used during stack creation for AWS stacks?

### Proposed solution

⚠️⚠️ REPLACE WITH PROPOSAL

**Comment by medikoo** [comment]
> This seems to make it difficult to debug stack creation when it fails.

@duboisj-illuminate thanks for report. What exactly debug information do you miss? AFAIK you can access deleted stacks in AWS CloudFormation console and inspect all the logs

**Comment by duboisj-illuminate** [comment]
Yes, very true: my initial report overstated the problem.  The logs are still available, yes.  As far as I can tell, serverless itself is not reporting the error information, but yes, it's still available in AWS.  Serverless appears to keep trying to query the stack and, when the stack has deleted, spits out an error because the stack is no longer present, like this: 

Serverless Error ---------------------------------------
 
  ServerlessError: Stack with id reportingservice-poc-test2-dev-duboisj-add-tests does not exist
      at /opt/atlassian/pipelines/agent/build/node_modules/serverless/lib/plugins/aws/provider/awsProvider.js:331:27
      at processTicksAndRejections (internal/process/task_queues.js:94:5)

Which at that point is true (the stack has been deleted), but serverless hasn't given a complete log of errors (even with SLS_DEBUG=*).  

But yes, things are still available in AWS, that is true.

**Comment by medikoo** [comment]
> Serverless Error ---------------------------------------
> 
> ServerlessError: Stack with id reportingservice-poc-test2-dev-duboisj-add-tests does not exist


@duboisj-illuminate in what scenario do you see that error? Also what exactly improvement you expect (sorry I'm getting a bit lost)
